### PR TITLE
[ci skip] Fix the page scrolling to top when in mobile mode

### LINKED ISSUE #53500: Opening edge guides with an anchor on mobile makes it impossible to scroll
For example on https://edgeguides.rubyonrails.org/configuring.html#config-action-view-field-error-proc it insists on scrolling the chapter guide into view:

https://github.com/user-attachments/assets/230a6fd7-7e32-43e8-8fc1-fb54ee836e26

When the guide is opened without an anchor, there is no problem.

Similarly, clicking on chapter sections results in the same behavior.

### PR DESCRIPTION
**Pull request by SleeplessByte**
### Motivation / Background

Fixes #53500 

### Checklist

Before submitting the PR make sure the following are checked:

* [X] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [X] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/assets/javascripts/guides.js [hunk 1]
```diff
@@ -52,6 +52,25 @@
       });
     }
 
+    // Detecting the mobile state should be done using CSS because that's what
+    // lays out the page, but this can fallback to JavaScript if that is not
+    // available
+    var MOBILE_WIDTH_FROM_STYLE_CSS_BREAKPOINT = 1024
+    var isMobile = window.innerWidth <= MOBILE_WIDTH_FROM_STYLE_CSS_BREAKPOINT;
+
+    if ('matchMedia' in window) {
```
Comment (MatheusRich) [inline guides/assets/javascripts/guides.js:61]: matchMedia seems pretty available. I wonder if the else here is necessary

**Comment by spencerchristensen** [comment]
Is there a reason `isMobile` is being assigned to `auto|scroll` and is not a simple boolean?

File: guides/assets/javascripts/guides.js (hunk 1 above)
Comment (SleeplessByte) [reply guides/assets/javascripts/guides.js:61]: A lot of the stuff is available but for some reason the guides are still using `var` among other things. So I've been adding Internet Explorer fallbacks for everything in the past as well.

Unless there comes a consensus guidance on what to support and not support I'm inclined to say that having more support is better than less.

Comment (p8) [reply guides/assets/javascripts/guides.js:61]: This PR has been merged already, but I don't think we need to support Internet Explorer.

Comment (SleeplessByte) [reply guides/assets/javascripts/guides.js:61]: I agree, but IMO there should be a consensus on what to support (`last 2 versions, not dead, > 0.2%` is common, but not accessible to all of the world and without a build pipeline not great perhaps for *documentation*), and that should be put inside the CONTRIBUTING page for the guides.

**Pull request merged**